prism/src/main/java/org/apache/falcon/resource/extensions/ExtensionManager.java

```diff
@@ -241,6 +256,11 @@ public APIResult delete(@PathParam("job-name") String jobName,
                             @DefaultValue("") @QueryParam("doAs") String doAsUser) {
         try {
             List<Entity> entities = getEntityList("", "", "", TAG_PREFIX_EXTENSION_JOB + jobName, "", doAsUser);
+            if (entities.size() == 0) {
+                // return failure if the extension job doesn't exist
+                return new APIResult(APIResult.Status.FAILED, "Extension job " + jobName + " doesn't exist.");
```

Comment:
Instead of returning APIResult.Status.FAILED, you should return 

APIResult(APIResult.Status.SUCCEEDED, entity + "(" + type + ") doesn't exist. Nothing to do");

Please refer to AbstractEntityManager